lib/tapioca/compilers/dsl/protobuf.rb

```diff
@@ -176,7 +191,8 @@ def field_of(descriptor)
             Field.new(
               name: descriptor.name,
               type: type,
-              init_type: type,
+              setter_type: descriptor.type == :enum ? "T.any(#{type}, Integer)" : type,
+              init_type: descriptor.type == :enum ? "T.any(#{type}, Integer)" : type,
```

Comment:
Only if optional is used. see my reply to the other comment. I think further changes are required to get this correct.

Reply:
edit: yes you are 100% right about the init type. I will make that change. I will have to test the setter though, because if optional is not used I'm not sure it can be set to nil with out a runtime error.

Reply:
irb> cart = Cart.new
=> <Cart: value_type: :ACROSS>
irb> cart.value_type = :ONE
=> :ONE
irb> cart.value_type = nil
Traceback (most recent call last):
....
Google::Protobuf::TypeError (Expected number or symbol type for enum field 'value_type'.)

Reply:
This is super confusing as the generated DSL uses a method called `optional`. However it does not mean it is optional in the context of setters/getters. if the new proto3 optional annotation is used (which I believe is still behind a feature flag) it will use proto3_optional in the DSL I believe